Lesson-9/SenderApp/SenderApp/CommonLogic/BussinessLogic/BussinessLogic.cs

```diff
@@ -0,0 +1,351 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Core.Abstractions.Layers;
+using Core.AppIdentity.Models;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Logging;
+using SenderApp.Data.RolesRepository;
+using SenderApp.Data.UsersRepository;
+using SenderApp.Models.DTO.Requests;
+using SenderApp.Models.DTO.Responses;
+using SenderApp.Models.ViewModels;
+using SenderApp.Models.ViewModels.LoginUser;
+
+namespace SenderApp.CommonLogic.BussinessLogic
+{
+    public interface IBussinessLayer :
+        IRegisterFacade<CustomIdentityUser, RequestFacadeDto>,
+        ILoginFacade<ResponseLoginDto, RequestFacadeDto>,
+        IRolesFacade<ResponseIdentityDto>,
+        IRolesEditFacade<ResponseChangeRoleDto>,
+        IUsersFacade<ResponseIdentityDto, RequestFacadeDto>,
+        IUsersEditFacade<ResponseEditUserDto>,
+        ISendMessageFacade
+    {
+    }
+
+    public class BussinessLogic : IBussinessLayer
+    {
+        private readonly UserManager<CustomIdentityUser> _userManager;
+        private readonly SignInManager<CustomIdentityUser> _signInManager;
+        private readonly ILogger<BussinessLogic> _logger;
+        private readonly IMapper _mapper;
+        private readonly IRolesRepository _rolesRepository;
+        private readonly IUserRepository _userRepository;
+        
+        public BussinessLogic(
+            UserManager<CustomIdentityUser> userManager, 
+            SignInManager<CustomIdentityUser> signInManager,
+            ILogger<BussinessLogic> logger,
+            IMapper mapper,
+            IRolesRepository rolesRepository,
+            IUserRepository userRepository)
+        {
+            _userManager = userManager;
+            _signInManager = signInManager;
+            _logger = logger;
+            _mapper = mapper;
+            _rolesRepository = rolesRepository;
+            _userRepository = userRepository;
+        }
+
+        /// <summary>
+        /// Фасад для регистрации новых пользователей
+        /// </summary>
+        /// <param name="requestDto"></param>
+        /// <returns></returns>
+        public async Task<CustomIdentityUser> RegisterFacadeAsync(string actionName, RequestFacadeDto requestDto)
+        {
+            _logger.LogInformation($"RegisterFacade start - {DateTime.Now}");
+            
+            if (requestDto != null)
+            {
```

Comment:
```suggestion
            if (requestDto != null)
                return null;
```

Старайтесь избегать большой вложенности условных блоков